src/runner.rs

```diff
@@ -135,14 +135,35 @@ impl RunHandle {
     }
 }
 
+/// Archival configuration
+#[derive(Clone)]
+pub enum ArchivalPolicy<Context> {
+    /// Archive nothing
+    None,
+    /// Archive all completed jobs
+    All,
```

Comment:
I was thinking about `Never` and `Always` as variant names. The idea was that this would be closer to how I think of policies: they always apply, or they never apply, or they apply under some condition. It would improve the readability in the method calls down below in my opinion.